rofi: 1.7.9.1 -> 2.0.0

SchweGELBin opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Resolves #439814
https://github.com/davatorium/rofi/releases/tag/2.0.0

Updated rofi to 2.0.0, which supports wayland now.
Dropped rofi-wayland and rofi-wayland-unwrapped
Added myself as a maintainer

It is a breaking change, so please let me know, what I should add to the release notes or whether/how I should add rofi-wyland back with a deprecation notice.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

MultisampledNight reviewed (commented):
Diff looks good!

wrvsrx reviewed (approved):
I test it on my machine and it works well.

MultisampledNight commented:
Belongs to `doc/release-notes/rl2511.section.md` imo, similar to e.g.

```
- `tooling-language-server` has been renamed to `deputy` (both the package and binary), following the rename of the upstream project.
```

MultisampledNight commented:
Should probably explicitly note that it was a merge

deejayem commented:
> The only problem I see is that people using rofi-wayland or the rofi-emoji-wayland need to switch to rofi or rofi-emoji. Do I need to note it somewhere?

Have a look in `pkgs/top-level/aliases.nix`.

SchweGELBin committed:
rofi: sort packages and inputs

SchweGELBin committed:
rofi: 1.7.9.1 -> 2.0.0

SchweGELBin committed:
rofi: add enable options for Wayland and X11 support

SchweGELBin committed:
rofi: add SchweGELBin as maintainer

SchweGELBin committed:
rofi-pass: drop rofi-wayland in favor of rofi

SchweGELBin committed:
rofi-wayland: drop in favor of rofi

rofi-wayland has been merged into rofi
https://github.com/davatorium/rofi/releases/tag/2.0.0

SchweGELBin committed:
rofi-emoji-wayland: drop in favor of rofi-emoji

rofi-wayland has been merged into rofi
https://github.com/davatorium/rofi/releases/tag/2.0.0

bew commented on doc/release-notes/rl-2511.section.md:
```diff
@@ -137,6 +137,10 @@
 
+- `rofi` has been updated to `2.0.0`. `rofi-wayland` has been merged into `rofi` and `rofi-wayland-unwrapped` has been merged into `rofi-unwrapped`. For more information and instructions see the [v2.0.0 changelog entry](https://github.com/davatorium/rofi/releases/tag/2.0.0).
+
+- `rofi` has been updated to `2.0.0`. `rofi-emoji-wayland` has been merged into `rofi-emoji`. For more information and instructions see the [v2.0.0 changelog entry](https://github.com/davatorium/rofi/releases/tag/2.0.0).
```
For this line I think it'd be clearer to start mentioning the change, then why it changed (easier to quick-read IMO).

Something like:
```suggestion
- `rofi-emoji-wayland` has been merged into `rofi-emoji` as `rofi` has been updated to `2.0.0` and supports both X11 & Wayland.
```

SchweGELBin replied:
What would be your suggested notes for `rofi-wayland` and `rofi-wayland-unwrapped`?

SchweGELBin replied:
And I think that the release notes should match the messages in aliases.nix, shouldn't they?

bew reviewed (commented):
Hello!
A few comments but overall it looks pretty good!

SchweGELBin commented on pkgs/applications/misc/rofi/default.nix:
```diff
@@ -47,44 +53,57 @@ stdenv.mkDerivation rec {
 
-  mesonFlags = [ "-Dimdkit=true" ];
+  mesonFlags =
+    lib.optionals withIMDKit [ "-Dimdkit=true" ]
```
Makes sense, I'll change remove this.

SchweGELBin commented on pkgs/tools/security/pass/rofi-pass.nix:
```diff
@@ -13,7 +13,6 @@
   gawk,
   gnused,
   # wayland-only deps
-  rofi-wayland,
```
Oops, I've missed it. I'll change the order.

SchweGELBin commented on pkgs/applications/misc/rofi/default.nix:
```diff
@@ -47,44 +53,57 @@ stdenv.mkDerivation rec {
+    libxkbcommon
+    pango
+    which
+    xcbutil
```
The package didn't build with `withWayland = true;` and `withX11 = false;` when I removed the `xcb*` packages.
Maybe I missed something? But packages that are only needed for X11 should of course be in an extra section.

SchweGELBin replied:
Hey, I've added them to an x11-specific section. libxkbcommon needs to support both backends; that is what I missed before.

SchweGELBin commented on pkgs/applications/misc/rofi/default.nix:
```diff
@@ -1,42 +1,48 @@
+  pandoc,
   pango,
+  pkg-config,
+  stdenv,
```
Alright, I'll change the order.
I'll do "lib, stdenv, ..., withWayland, withX11".

SchweGELBin replied:
Should I change e.g. `withWayland` to something like `waylandSupport`?
Or would it be better to have an enum like `backend = [ "both" "wayland" "x11"]`?

SchweGELBin replied:
I've used [this](https://github.com/davatorium/rofi/blob/2.0.0/INSTALL.md) as a guide.

> Be default rofi builds with both backends (x11 and wayland) if available on the system. If no backend is found, it will give an error. You can force the build system to disable the [wayland](https://github.com/davatorium/rofi/blob/2.0.0/INSTALL.md#disable-wayland-support) or [x11](https://github.com/davatorium/rofi/blob/2.0.0/INSTALL.md#disable-x11-support) backend.

This is the reason, why I included both options. The user can disable a backend or leave the default options that support both backends.

bew reviewed (approved):
LGTM 😎👌

Mange reviewed (approved):
Well done! :heart: 

(Disclaimer: I only reviewed commits that related to `rofi-emoji`.)

SigmaSquadron commented on doc/release-notes/rl-2511.section.md:
```diff
@@ -137,6 +137,10 @@
 
 - `pulsemeeter` has been updated to `2.0.0`. The configuration file from older versions has to be deleted. For more information and instructions see the [v2.0.0 changelog entry](https://github.com/theRealCarneiro/pulsemeeter/releases/tag/v2.0.0).
 
+- `rofi` has been updated to `2.0.0`. `rofi-wayland` and `rofi-wayland-unwrapped` have been merged into `rofi` and `rofi-unwrapped` respectively. For more information and instructions see the [v2.0.0 changelog entry](https://github.com/davatorium/rofi/releases/tag/2.0.0).
```
Are there major backwards incompatibilities on 2.0.0? This doesn't seem like it needs a changelog entry in the Nixpkgs manual.
